src/shared/capture/capture_splitter.cpp

```diff
@@ -0,0 +1,188 @@
+      settings->addChild(relative_height_offset = new VarDouble("Relative height offset", 0.4, 0.0, 1.0));
+      settings->addChild(relative_width_offset = new VarDouble("Relative width offset", 0.0, 0.0, 1.0));
+      break;
+    case 2:
+      settings->addChild(relative_height_offset = new VarDouble("Relative height offset", 0.0, 0.0, 1.0));
+      settings->addChild(relative_width_offset = new VarDouble("Relative width offset", 0.4, 0.0, 1.0));
+      break;
+    case 3:
+      settings->addChild(relative_height_offset = new VarDouble("Relative height offset", 0.4, 0.0, 1.0));
+      settings->addChild(relative_width_offset = new VarDouble("Relative width offset", 0.4, 0.0, 1.0));
+      break;
+  }
+
+  if(default_camera_id < 4) {
+    settings->addChild(relative_height = new VarDouble("Relative height", 0.6, 0.0, 1.0));
+    settings->addChild(relative_width = new VarDouble("Relative width", 0.6, 0.0, 1.0));
+  } else {
+    settings->addChild(relative_height = new VarDouble("Relative height", 1.0, 0.0, 1.0));
+    settings->addChild(relative_width = new VarDouble("Relative width", 1.0, 0.0, 1.0));
+  }
+}
+
+bool CaptureSplitter::stopCapture()
+{
+  cleanup();
+  return true;
+}
+
+void CaptureSplitter::cleanup()
+{
+#ifndef VDATA_NO_QT
+  mutex.lock();
+#endif
+  is_capturing=false;
+  full_image_arrived_mutex.unlock();
+  frame_processed_mutex.unlock();
+#ifndef VDATA_NO_QT
+  mutex.unlock();
+#endif
+}
+
+bool CaptureSplitter::startCapture()
+{
+#ifndef VDATA_NO_QT
+  mutex.lock();
+#endif
+
+  is_capturing = true;
+  full_image_arrived_mutex.try_lock();
+  frame_processed_mutex.try_lock();
+
+#ifndef VDATA_NO_QT
+  mutex.unlock();
+#endif
+  return true;
+}
+
+bool CaptureSplitter::copyAndConvertFrame(const RawImage &src, RawImage &target) {
+#ifndef VDATA_NO_QT
+  mutex.lock();
+#endif
+
+  if (src.getWidth() == 0 || src.getHeight() == 0) {
+#ifndef VDATA_NO_QT
+    mutex.unlock();
+#endif
+    return false;
+  }
+  target.setTime(src.getTime());
+
+  // make sure we use values from 0 to 1
+  double rel_height_offset = std::min(1.0, std::max(0.0, relative_height_offset->getDouble()));
+  double rel_height = std::min(1.0, std::max(0.0, relative_height->getDouble()));
+  double rel_width_offset = std::min(1.0, std::max(0.0, relative_width_offset->getDouble()));
+  double rel_width = std::min(1.0, std::max(0.0, relative_width->getDouble()));
+
+  // make sure we keep within the src image
+  rel_height = std::min(rel_height, 1 - rel_height_offset);
+  rel_width = std::min(rel_width, 1 - rel_width_offset);
+
+  // calculate offset in src image
+  int height_offset = (int) (src.getHeight() * rel_height_offset);
+  int width_offset = (int) (src.getWidth() * rel_width_offset);
+
+  // calculate dimensions of target image
+  int height = (int) (src.getHeight() * rel_height);
+  int width = (int) (src.getWidth() * rel_width);
+
+  // for some reason, the image gets screwed when the width is not a multiple of 4
+  width -= width % 4;
```

Comment:
@joydeep-b Any idea, why images get screwed with certain width? Looks like it's mostly fine with this hack, but I already saw another glitch with this hack (colors have changed).
We already experienced this when changing the AOI of the bluefox cameras (which is directly set in the cameras). So I do not expect this is an issue during conversion, but rather during visualization.
The visualizations, like field calibration result, is screwed as well.

Example:
![2019-01-14-215712_414x369_scrot](https://user-images.githubusercontent.com/779094/51140657-681d3500-1847-11e9-887f-9c0cea4a6d19.png)

Reply:
OpenGL Texture operations require specific alignment. The easiest way to do this is to ensure that the image dimensions are in multiples of 4. Or else, you'll need to manually create a new image texture with padding to the next multiple of 4. See:
https://www.khronos.org/opengl/wiki/Common_Mistakes#Texture_upload_and_pixel_reads

Reply:
oh, I see. Thx for the hint :)